src/services/appointment.service.ts

```diff
@@ -31,74 +41,145 @@ export default class AppointmentService {
 
-    this.existingAppointment(scheduledStart);
+    this.validateAppointment(scheduledStart);
+    await this.ensureNoExistingAppointment(scheduledStart);
 
-    this.appointmentValidation(scheduledStart);
+    const customer = await this.getOrCreateCustomer(createDto);
 
-    let customer = await this.customerRepository.findOne({
-      where: { phone: createDto.phone },
-    });
-    if (!customer) {
-      const customerDto = new CustomerRegisterDto();
-      customerDto.name = createDto.name;
-      customerDto.phone = createDto.phone;
-      customer = await this.customerService.createCustomer(customerDto);
-    }
-
-    const totalPrice = treatments.reduce(
-      (sum, treatment) => sum + Number(treatment.price),
-      0,
-    );
+    const totalPrice = this.calculateTotalPrice(treatments);
 
     const appointment = this.appointmentRepository.create({
       status: Status.PENDING,
       scheduled_start: scheduledStart,
       total_price: totalPrice,
       duration: serviceDuration,
-      user: user,
-      customer: customer,
-      treatments: treatments,
+      user,
+      customer,
+      treatments,
     });
 
     try {
-      const saved = await this.appointmentRepository.save(appointment);
-      const messageText = messages['appointment_reminder'];
-      await this.whatsAppService.sendInteractiveMessage(
-        customer.phone,
-        messageText,
+      const savedAppointment = await this.appointmentRepository.save(
+        appointment,
       );
-      return saved;
+
+      try {
+        await this.sendAppointmentConfirmationMessage(
+          customer.phone,
+          scheduledStart,
+          treatments,
+        );
+        this.scheduleReminderMessage(savedAppointment);
+      } catch (error) {
+        this.logger.error(`Error al conectar con whapi: ${error.message}`);
+      }
+
+      return savedAppointment;
     } catch (error) {
       throw new BadRequestException(
         `Error creating appointment: ${error.message}`,
       );
     }
   }
 
+  async getLastAppointmentByPhone(phone: string): Promise<Appointment> {
+    try {
+      const customer = await this.customerRepository.findOne({
+        where: { phone },
+      });
+
+      if (!customer) {
+        return null;
+      }
+
+      return this.appointmentRepository.findOne({
+        where: { customer },
+        order: { id: 'DESC' },
+        relations: ['treatments'],
+      });
+    } catch (error) {
+      throw new BadRequestException(
+        `Error al obtener la última cita: ${error.message}`,
+      );
+    }
+  }
+
+  scheduleReminderMessage(appointment: Appointment) {
+    try {
+      const reminderTime = new Date(appointment.scheduled_start);
+      reminderTime.setHours(reminderTime.getHours() - 12);
+
+      const now = new Date();
+      if (reminderTime <= now) {
+        this.logger.log(
+          `Tiempo excedido para el recordatorio de la cita: ${appointment.id}.`,
+        );
+        return;
```

Comment:
💭 if you're being explicit in this line about returning [null](https://github.com/cmoreno357/mindticas2025-back/pull/26/files#diff-ffade9a36b9a2d53d1cecc3eebc25f8abf4f715f8dc259e887fe3a30d23ea446R97)

shouldn't be more explicit here to follow the same standard? 🤔 

unless we want to remove the null in line 97